### DISCUSSION TITLE
Is the HEIC support considered safe from a patent perspective?

### DISCUSSION THREAD

**Discussion by StefanOltmann** [drewnoakes/metadata-extractor#646] [Q&A] [answered]
HEIC is widely acknowledged to be heavily burdened by patents.

Is it safe to utilize this library for extracting metadata from iPhone HEIC files, or is it necessary to obtain a license first?

Do projects employing this library need to secure a license, even if they do not read HEIC files at all?

From my observation, it seems to present significant challenges.

Could you please provide insights into the legal standing in this context?

**Comment by drewnoakes** [comment, marked as answer]
> Could you please provide insights into the legal standing in this context?

I'm not a lawyer and can't justify paying one to answer this question definitely. The library is offered without any warranty, as specified in the license, so you have to make your own decision here.

My understanding is that it's the HEVC video codec that's covered by patents, not the container format (which is the standard ISO 14496, aka ISO base media file format, used for MP4 and others), nor the metadata format (which isn't anything specific to HEIC or HEVC).

**Reply by StefanOltmann** [reply]
Thank you, Drew!

I agree with your insight. It seems that the patents for ISOBMFF have likely expired.

Regarding my library metadata parsing library at https://github.com/Ashampoo/kim, where I initially ported all logic from your "heif" package, I've decided to explicitly name it "isobmff", remove all non-ISOBMFF covered code, and let the logic handle files detected as HEIC.

Upon reviewing the original ISO 14496 document, I discovered that the containers for image size ("ispe") and image rotation ("irot") are not part of ISOBMFF but specified in ISO 23008, specifically for HEIC.

This information is also documented in your Javadoc: https://github.com/drewnoakes/metadata-extractor/blob/e24a17bd0a2b76380d655a2e1c40843e557e21bd/Source/com/drew/metadata/heif/boxes/ImageSpatialExtentsProperty.java#L29

While it might be safe to use, I am not willing to take the risk of potential patent infringement and have removed them and everything else specific to HEIC (expect the magic bytes) from my library.

Thank you for creating metadata-extractor, and I appreciate your advice!
